helix-view/src/document.rs

```diff
@@ -940,14 +955,46 @@ impl Document {
         self.undo_redo_impl(view, false)
     }
 
-    pub fn savepoint(&mut self) {
-        self.savepoint = Some(Transaction::new(self.text()));
+    /// Creates a savepoint. Until `doc.restore` is called,
+    ///
+    /// `Document::savepoint` is set to `None`, or `savepoint` is called again
+    /// a transaction to revert the document back to this savepoint is automatically
+    /// maintained. `Document::restore` can be used to reset the document contents
+    /// and view selection back to the state when this function was called.
+    ///
+    /// # Returns
+    ///
+    /// An `id` that can uniquely identify this savepoint. A document
+    /// can only track a single savepoint at a time, to check if your savepoint still
+    /// applies simply
```

Comment:
Looks like the sentence was never finished

Reply:
Ah this documentation is actually outdated and refers to an old version of this function from a previous slightly different approach that I moved awaybfrom (refcount stored on the document, and savepoints tracked using ids, was very error prone because it was easy to forgot to drop a savepoint or use a dropped savepoint for a revert, the recounting is much more robust). Will fix that comment in a bit